crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Strings.hs

```diff
@@ -3,27 +3,261 @@
 {-# OPTIONS_GHC -Wno-warnings-deprecations #-}
 
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveFunctor #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE ImplicitParams #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections #-}
 {-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
 
 -- | Manipulating C-style null-terminated strings
 module Lang.Crucible.LLVM.MemModel.Strings
   ( Mem.loadString
   , Mem.loadMaybeString
   , Mem.strLen
+  , loadConcretelyNullTerminatedString
+  , loadSymbolicString
   , storeString
   ) where
 
+import           Data.Bifunctor (Bifunctor(bimap, first))
+import           Control.Monad.IO.Class (MonadIO, liftIO)
+import qualified Data.BitVector.Sized as BV
+import           Data.Functor ((<&>))
 import qualified Data.Parameterized.NatRepr as DPN
+import           Data.Word (Word8)
 import qualified Data.Vector as Vec
+import qualified GHC.Stack as GHC
 import qualified Lang.Crucible.Backend as LCB
+import qualified Lang.Crucible.Backend.Online as LCBO
+import qualified Lang.Crucible.Simulator as LCS
 import qualified Lang.Crucible.LLVM.DataLayout as CLD
 import qualified Lang.Crucible.LLVM.MemModel as LCLM
 import qualified Lang.Crucible.LLVM.MemModel as Mem
+import qualified Lang.Crucible.LLVM.MemModel.Partial as Partial
+import qualified What4.Expr.Builder as WEB
 import qualified What4.Interface as WI
+import qualified What4.Protocol.Online as WPO
+import qualified What4.SatResult as WS
+
+-- | Whether to stop or keep going
+--
+-- Like Rust's @std::ops::ControlFlow@.
+data ControlFlow a b
+  = Continue a
+  | Break b
+  deriving Functor
+
+instance Bifunctor ControlFlow where
+  bimap l r =
+    \case
+      Continue x -> Continue (l x)
+      Break x -> Break (r x)
```

Comment:
Given that `ControlFlow` isn't exported, is there any reason to define this instance? As far as I can tell, nothing is making use of it.